chaos_genius/controllers/kpi_controller.py

```diff
@@ -49,46 +42,33 @@ def get_kpi_data_from_id(n: int) -> dict:
 
 
 def run_anomaly_for_kpi(
-    kpi_id: int,
-    end_date: date = None,
-    task_id: Optional[int] = None
-) -> Union["typing.Literal[False]", date]:
-
-    try:
-        logger.info(f"Starting Anomaly Detection for KPI ID: {kpi_id}.")
-        kpi_info = get_kpi_data_from_id(kpi_id)
-        logger.info("Retrieved KPI information.")
-
-        logger.info("Selecting end date.")
+    kpi_id: int, end_date: Optional[date] = None, task_id: Optional[int] = None
+):
 
-        if end_date is None and kpi_info["scheduler_params"]["scheduler_frequency"] == "D":
-            # by default we always calculate for n-days_offset_for_analytics
-            end_date = datetime.today().date() - timedelta(days=(DAYS_OFFSET_FOR_ANALTYICS))
-            # Check if data is available or not then try for n-days_offset_for_analytics-1
-            if not _is_data_present_for_end_date(kpi_info, end_date):
-                end_date = end_date - timedelta(days=1)
-                logger.info("Decreasing end date by 1.")
+    logger.info(f"Starting Anomaly Detection for KPI ID: {kpi_id}.")
+    kpi_info = get_kpi_data_from_id(kpi_id)
+    logger.info("Retrieved KPI information.")
 
-        elif end_date is None and kpi_info["scheduler_params"]["scheduler_frequency"] == "H":
-            end_date = datetime.today().date()
+    logger.info("Selecting end date.")
 
-        logger.info(f"End date is {end_date}.")
+    if end_date is None and kpi_info["scheduler_params"]["scheduler_frequency"] == "D":
+        # by default we always calculate for n-days_offset_for_analytics
+        end_date = datetime.today().date() - timedelta(days=(DAYS_OFFSET_FOR_ANALTYICS))
+        # Check if data is available or not then try for n-days_offset_for_analytics-1
+        if not _is_data_present_for_end_date(kpi_info, end_date):
+            end_date = end_date - timedelta(days=1)
+            logger.info("Decreasing end date by 1.")
```

Comment:
are these kinds of log useful in prod instances as not much context being logged

Reply:
These are useful, but yes, we should add more context to these (currently, context is inferred from the surrounding logs)